### PR TITLE
Changes to UsersScoreWriter - MOAB writer for G410.3 support

### PR DESCRIPTION
**Pull request by makeclean**
This PR adds changes to the UerScoreWriter class due to the changes in the current version of Geant4. The #if statements allows pre 10.3 code to live alongside the 10.2 relevant stuff. Addresses #510

### CODE REVIEW

**Comment by ljacobson64** [comment]
@makeclean Can you fix the conflict? I'll merge afterwards.

File: geant4/build/src/ExN01UserScoreWriter.cc [hunk 1]
```diff
@@ -178,8 +192,6 @@ moab::ErrorCode ExN01UserScoreWriter::generate_moab_mesh(std::vector<double> x_b
       }
     }
   }
-  rval = MBI()->write_mesh("halfway.h5m");
```
Comment (ljacobson64) [inline geant4/build/src/ExN01UserScoreWriter.cc:181]: do we want this to be deleted?

Comment (makeclean) [reply geant4/build/src/ExN01UserScoreWriter.cc:181]: Yes, its a lingering debug thing, shouldve been gone long ago

**Comment by ljacobson64** [comment]
looks good

**Pull request merged**